runtime/lua/vim/treesitter/_query_linter.lua

```diff
@@ -36,6 +36,7 @@ local function add_lint_for_node(diagnostics, node, buf, lint, lang)
   local message = lint .. ': ' .. node_text
   local error_range = { node:range() }
   diagnostics[#diagnostics + 1] = {
+    bufnr = buf,
```

Comment:
is this a bug fix?

Reply:
It's not a bug fix as setting `bufnr` is not required (nor encouraged). These diagnostics are eventually passed to `vim.diagnostic.set()`. The buffer is set there.

`:h diagnostic-structure` indicates which fields are required. The type annotations should be updated to reflect that.

Reply:
fixed!